File: 10173.cpp
#include <iostream>
#include <string>
using namespace std;

bool found_nemo(string s) {
	string nemo[8] = { "EMO", "EMo", "EmO", "eMO", "Emo", "eMo", "emO", "emo" };
	for (int i = 0; i < s.length() - 3; i++) {
		if (s[i] == 'N' || s[i] == 'n') {
			for (int j = 0; j < 8; j++)
				if (s.substr(i + 1, 3) == nemo[j])
					return true;
		}
	}

	return false;
}

int main() {
	while (1) {
		string s; getline(cin, s);
		if (s == "EOI") break;

		if (found_nemo(s))
			cout << "Found\n";
		else
			cout << "Missing\n";
	}
}

File: 10173.py
def found_nemo(s):
    nemo = ["EMO", "EMo", "EmO", "eMO", "Emo", "eMo", "emO", "emo"]
    for i in range(len(s) - 3):
        if s[i] == 'N' or s[i] == 'n':
            if s[i+1:i+4] in nemo:
                return True

    return False

while 1:
    s = input()
    if s == 'EOI': break

    if found_nemo(s):
        print("Found")
    else:
        print("Missing")
